pandas/core/indexes/numeric.py

```diff
@@ -72,6 +74,19 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None, fastpath=None):
             name = data.name
         return cls._simple_new(subarr, name=name)
 
+    def _validate_dtype(cls, dtype):
+        if not dtype:
+            return
+        if cls._typ == "int64index":
+            if not is_signed_integer_dtype(dtype):
+                raise ValueError("Incorrect `dtype` passed")
```

Comment:
Can you add a little more info to these messages, like what is expected and what was received?